### ISSUE TITLE
MA0151 false positive

### ISSUE THREAD

**Issue by dmitriy-maksimov-wd** [meziantou/Meziantou.Analyzer#831]
### Version of the Meziantou.Analyzer NuGet package

2.0.216

### Rule Identifier

MA0151

### Target Framework

net8.0

### C# Language version

C# 11

### Description

I have `IList<>` member.
Getting "Member `Count` does not exist", but the member exists in `ICollection` (which `IList<>` derived from).
The recent fix #830 seems to broke it.

### Reproduction Steps

Minimal code:

```c#
[DebuggerDisplay("{referenceString} ({allStrings.Count})")]
public struct ContainedStrings
{
    private readonly IList<string> allStrings;
    private readonly string referenceString;

    public void Test()
    {
        var allStringsCount = allStrings.Count;
    }
}
```

**Comment by meziantou** [comment]
It should be fixed by 7fd50604f025810f531020e1d61795060e6c72af. The new version will be available very soon on nuget.org (https://www.nuget.org/packages/Meziantou.Analyzer/2.0.217).

**Comment by luispfgarces** [comment]
Hi @meziantou,

It seems that it is still raising false positives when the type of a property is a generic type parameter.

Here's my specific scenario:

```
[DebuggerDisplay("{RightOperand.DataType.ToString(),nq} condition: <{Condition.ToString(),nq}> {Operator} {RightOperand.Value}")]
public class ValueConditionNode<TCondition> : IValueConditionNode<TCondition>
    where TCondition : notnull
{
    private readonly ValueConditionNode valueConditionNode;

    public ValueConditionNode(ValueConditionNode valueConditionNode)
    {
        this.valueConditionNode = valueConditionNode;
    }

    public TCondition Condition => GenericConversions.Convert<TCondition>(valueConditionNode.Condition);

    public LogicalOperators LogicalOperator => LogicalOperators.Eval;

    public Operators Operator => this.valueConditionNode.Operator;

    public IDictionary<string, object> Properties => this.valueConditionNode.Properties;

    public Operand RightOperand => this.valueConditionNode.RightOperand;
}
```

It raises the diagnostic message:

> Member 'ToString' does not exist (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0151.md)

If we remove the `ToString()` from the `Condition`, the diagnostic is resolved, but it is still raised if we remove it from the `RightOperand.DataType`.

Hope this helps with troubleshooting the issue. Thank you for your time!

**Comment by meziantou** [comment]
Thanks for reporting this! Next time, could you please open a new issue instead of replying to a closed one? It makes it easier for me to keep track of things.

I push new changes (358803cb5df589db80a0e1b94e72340151494491) that should detect the type parameter case, and also special case `ToString`. 
https://www.nuget.org/packages/Meziantou.Analyzer/2.0.218

**Comment by luispfgarces** [comment]
Will do next time 👍 Thank you for solving this!
